[ZiliaAJY] iP

ZiliaAJY opened the pull request.

ZiliaAJY committed:
Add rename, greet, exit features

ZiliaAJY committed:
Add rename, greet, exit features

ZiliaAJY committed:
Add 'Echo', 'Add, List', 'Mark as Done' features, and Task class

- Implement Task class for managing tasks
- Add Conglo$Task.class file and Conglo.class file

ZiliaAJY committed:
Align code with Java coding standards

- Refined method and variable names for consistency and clarity
- Adjusted layout and spacing for improved readability

ZiliaAJY committed:
Add Todo, Event, and Deadline features

- Add 'Task.java', 'Deadline.java', 'Todo.java' and 'Event.java' files
- Implement classes Todo, Deadline and Event classes to inherit from a Task class

ZiliaAJY committed:
Refactor code and update names; add JavaDoc comments

- Refactored method names and constants for improved clarity
- Added JavaDoc comments to Task, Deadline, Event, and Todo classes
  - Included class-level documentation and descriptions for constructors and methods.
  - Documented parameters, return values, and method behaviors

Chiang-HuiXin commented on src/main/java/Conglo.java:
```diff
@@ -0,0 +1,168 @@
+            markTask(words, taskList);
+            break;
+        case "todo":
+            Todo todo = new Todo(words[1]);
+            taskList[taskCount] = todo;
+            taskCount++;
```
Maybe put this in a method?

Chiang-HuiXin commented on src/main/java/Conglo.java:
```diff
@@ -0,0 +1,168 @@
+     *
+     * @param args Command-line arguments (not used).
+     */
+    public static void main(String[] args) {
```
I like how your main method is short!

Chiang-HuiXin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,47 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public void markAsDone() {
```
Maybe name it as "setMarkAsDone" since it is a setter?

Chiang-HuiXin commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+        this.by = by;
+    }
+
+    /**
+     * Returns a string representation of the Deadline task.
+     * The format includes a label for deadlines, the task's completion status,
+     * description, and the deadline date.
+     *
+     * @return A string representation of the Deadline task.
+     */
```
I like that you do documentation, it really makes the code much more readable!

Chiang-HuiXin reviewed (commented):
Good job! Overall, I think that your code is very neat and readable. There are good abstractions and documentation.

seanngja commented on src/main/java/Conglo.java:
```diff
@@ -0,0 +1,168 @@
+        sayGoodbye();
+    }
+
+    /**
+     * Prints a line separator to the console.
+     */
```
I like how you have added comments that describe what each function does.

seanngja commented on src/main/java/Conglo.java:
```diff
@@ -0,0 +1,168 @@
+     *
+     * @param args Command-line arguments (not used).
+     */
+    public static void main(String[] args) {
+        greetUser();
+        Scanner scanner = new Scanner(System.in);
+        Task[] listing = new Task[MAX_TASKS];
+        command = scanner.nextLine();
+        while (!command.equals("bye")) {
+            processCommand(command, listing);
+            command = scanner.nextLine();
+        }
+        scanner.close();
+        sayGoodbye();
+    }
```
I like how your main function is readable and easy to follow!

seanngja commented on src/main/java/Conglo.java:
```diff
@@ -0,0 +1,168 @@
+     */
+    public static void processCommand(String command, Task[] taskList) {
+        String[] words = command.split(" ", 2);
+        switch(words[0]) {
```
Your switch case indentation looks good, great job!

seanngja commented on src/main/java/Conglo.java:
```diff
@@ -0,0 +1,168 @@
+            echoTask(todo);
+            break;
+        case "deadline":
+            addDeadline(words[1], taskList);
```
I like that your functions are named concisely and easy to understand.

ZiliaAJY committed:
Update input.txt and expected.txt for testing scenarios

- Renamed paths in runtest.bat
- Added new tasks and commands to test various functionalities, including:
  - Adding and listing tasks
  - Marking tasks as done
  - Handling multiple commands in sequence

ZiliaAJY committed:
Update comments and remove typos

ZiliaAJY committed:
Add "CongloException.java" file

-Contains Conglo's custom exception including unknown command, missing description, invalid or missing task number and invalid format

ZiliaAJY committed:
Add and Refactor Exception handling

- Use base CongloException for common exception handling
- Added internal try-catch block to handle specific exceptions
- Improved error handling and message output

ZiliaAJY committed:
Merge branch 'branch-Level-5'

ZiliaAJY committed:
Update JavaDoc comments to include exception details

ZiliaAJY committed:
Add "QuickManual.java" file to print instructions manual

ZiliaAJY committed:
Organize classes into packages and update Conglo.java to use them

- Created packages "conglo", "conglo,exception", "conglo.manual" and "conglo.task"

ZiliaAJY committed:
Merge branch 'branch-A-Packages'

ZiliaAJY committed:
Implement automatic task saving and loading from disk

- Save tasks automatically to the hard disk whenever the task list is modified.
- Load tasks from the hard disk when the application starts, ensuring persistence across sessions.
- Hard-coded file path and name for storage: './data/conglo.txt'.
- Added error handling for file I/O operations to manage file creation and loading processes.

ZiliaAJY committed:
Implement automatic task saving and loading from disk

- Save tasks automatically to the hard disk whenever the task list is modified.
- Load tasks from the hard disk when the application starts, ensuring persistence across sessions.
- Hard-coded file path and name for storage: './data/conglo.txt'.
- Added error handling for file I/O operations to manage file creation and loading processes.

ZiliaAJY committed:
Replace arrays with ArrayLists and add task deletion

- Replaced all instances of normal arrays with ArrayLists for better flexibility.
- Implemented functionality to delete tasks from the list.

ZiliaAJY committed:
Merge branch 'branch-Level-6'
Add task deletion functionality

- Implement support for deleting tasks from the list.
- Update task management to handle deletions correctly.
- Ensure task list updates reflect changes immediately.

ZiliaAJY committed:
Resolved merge conflicts between branch-Level-6 and branch-Level-7

- Resolved conflicts in Task.java
- Resolved conflicts in Conglo.java

Description of changes made to resolve conflicts:
- Fixed discrepancies in file paths and format between branches
- Refactor Task class and its children class to adopt new format for saving of tasks

ZiliaAJY committed:
Fix runtest.bat; update input/expected files; handle "list" exception

ZiliaAJY committed:
Fix symbol display

ZiliaAJY committed:
Resolved merge conflict between master and branch-Level-7

ZiliaAJY committed:
Merge branch 'branch-Level-7'
Merge branch-Level-7 to include Save function

okkhoy commented on src/main/java/conglo/Conglo.java:
```diff
@@ -0,0 +1,253 @@
+package conglo;
+
+import conglo.manual.*;
```
don't use wildcard imports

okkhoy commented on src/main/java/conglo/Conglo.java:
```diff
@@ -0,0 +1,253 @@
+ */
+public class Conglo {
+
+    // Constants
```
redundant comment

okkhoy commented on src/main/java/conglo/Conglo.java:
```diff
@@ -0,0 +1,253 @@
+        } else {
+            System.out.println("All done! Task added to list:");
+        }
+        String taskSuffix = size > 1 ? " tasks" : " task";
```
could consider making this "easy to read" by using explicit `if ... else` block.

okkhoy commented on src/main/java/conglo/Conglo.java:
```diff
@@ -0,0 +1,253 @@
+     * @throws CongloException.InvalidFormat If the deadline format is incorrect.
+     */
+    public static void addDeadline(String sentence) throws CongloException.InvalidFormat {
+        if (!sentence.contains(" /by ")) {
```
can try improving SLAP in such cases

okkhoy commented on src/main/java/conglo/Conglo.java:
```diff
@@ -0,0 +1,253 @@
+    }
+
+    public static void deleteTask(String word) throws CongloException.InvalidTaskNumber {
+        int i;
```
use a more descriptive variable name.

okkhoy commented on src/main/java/conglo/Conglo.java:
```diff
@@ -0,0 +1,253 @@
+            }
+            listTasks();
+            break;
+        case "unmark":
```
fall through comment is needed

okkhoy commented on src/main/java/conglo/exception/CongloException.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    // Custom exception for unknown commands
+    public static class UnknownCommand extends CongloException {
```
consider making these separate classes rather than nested classes

okkhoy commented on src/main/java/conglo/storage/TaskStorage.java:
```diff
@@ -0,0 +1,85 @@
+    private static final String FILE_PATH = "./data/conglo.txt";
+    private static final String DIRECTORY_PATH = "./data";
+
+    public static ArrayList<Task> loadTasks() {
```
you can improve SLAP here

okkhoy commented on src/main/java/conglo/storage/TaskStorage.java:
```diff
@@ -0,0 +1,85 @@
+                case "D":
+                    if (parts.length < 4) {
+                        System.out.println("Invalid deadline format: " + line);
+                        continue;
```
is there a way to remove arrow head code here?

okkhoy reviewed (commented):
There are a few coding standards and code quality violations in the code. Please try to fix them before the final submission.

ZiliaAJY committed:
Extract code into Ui, Storage, Parser, and TaskList classes

- Created the Ui class for user interface interactions and message displays
- Implemented the Storage class for task storage and file management
- Introduced the Parser class to handle command parsing and user input processing
- Created the TaskList class to enhance task management functionalities
- Improved code organization and readability

ZiliaAJY committed:
Clean up typos

ZiliaAJY committed:
Refactor extracted classes

- Removed use of ArrayList<> in Parser.java
- Cleaned up code for readability

ZiliaAJY committed:
Implement date parsing for Deadlines

- Added DateParser class to handle date and time input for deadlines
- Updated Deadline class to store deadline dates as LocalDate and format output to "dd MMM yyyy"
- Enhanced user input acceptance for dates in dd-MM-yyyy HHmm format
- Implemented conversion of 24-hour time format to am/pm in deadline outputs
- Update Manual.java to include the newly accepted format

ZiliaAJY committed:
Implement Task Search Feature

- Added a method to search for tasks by keyword within the TaskList class
- Created a findTasks() method that returns a list of matching tasks
- Updated the Parser class to handle user commands for finding tasks
- Updated Manual.java to include instruction for finding tasks

ZiliaAJY committed:
Add Javadoc comments to all classes and methods for comprehensive documentation

- Implemented header comments for all public and non-private classes.
- Added Javadoc comments to all non-trivial methods, enhancing clarity.

ZiliaAJY committed:
Merge pull request #1 from ZiliaAJY/branch-Level-8

Branch Level 8

ZiliaAJY committed:
Resolved merge conflicts with master

ZiliaAJY committed:
Resolved merge conflicts with master

ZiliaAJY committed:
Merge pull request #2 from ZiliaAJY/branch-Level-9

Branch Level 9

ZiliaAJY committed:
Resolved merge conflicts with master

ZiliaAJY committed:
Merge pull request #3 from ZiliaAJY/branch-A-JavaDoc

Add Javadoc comments to all classes and methods for comprehensive doc…

ZiliaAJY committed:
Update test case to include 'Date' and 'Find' features

ZiliaAJY committed:
Update README.md: Add features, commands, prerequisites, and setup instructions for Conglo Task Manager Chatbot

ZiliaAJY committed:
Clean up code formatting and update comments for clarity

ZiliaAJY committed:
Refactor code and update comments for clarity

- Remove arrow head code

ZiliaAJY committed:
Merge branch 'master' of https://github.com/ZiliaAJY/ip

ZiliaAJY committed:
Add method to create file if it does not exist

ZiliaAJY committed:
Fix parsing int bug in mark task

ZiliaAJY committed:
Reorganise README files
